yt_dlp/extractor/nintendo.py

```diff
@@ -1,57 +1,101 @@
+            'upload_date': '20230831',
+            'age_limit': 6,
         },
-        'add_ie': ['Ooyala'],
+    }, {
+        'url': 'https://www.nintendo.com/en/nintendo-direct/09-04-2019/',
+        'only_matching': True,
     }]
 
+    def _create_asset_url(self, path):
+        return urljoin('https://assets.nintendo.com/', quote(path))
+
     def _real_extract(self, url):
-        page_id = self._match_id(url)
+        locale, video_id = self._match_valid_url(url).group('locale', 'id')
+
+        language, _, country = (locale or '').partition('-')
+        parsed_locale = f'{language.lower() if language else "en"}_{country.upper() if country else "US"}'
+        self.write_debug(f'Using locale {parsed_locale} (from {locale})', only_once=True)
+
+        response = self._download_json('https://graph.nintendo.com/', video_id, query={
+            'operationName': 'NintendoDirect',
+            'variables': json.dumps({
+                'locale': parsed_locale,
+                'slug': video_id,
+            }, separators=(',', ':')),
+            'extensions': json.dumps({
+                'persistedQuery': {
+                    'version': 1,
+                    'sha256Hash': '969b16fe9f08b686fa37bc44d1fd913b6188e65794bb5e341c54fa683a8004cb'
+                },
+            }, separators=(',', ':')),
+        })
+        # API returns `{"data": {"direct": null}}` if no matching id
+        direct_info = traverse_obj(response, ('data', 'direct', {dict}))
+        if not direct_info:
+            raise ExtractorError(f'No Nintendo Direct with id {video_id} exists', expected=True)
 
-        webpage = self._download_webpage(url, page_id)
+        errors = ', '.join(traverse_obj(response, ('errors', ..., 'message')))
+        if errors:
+            raise ExtractorError(f'GraphQL API error: {errors or "Unknown error"}')
 
-        entries = [
-            OoyalaIE._build_url_result(m.group('code'))
-            for m in re.finditer(
-                r'data-(?:video-id|directVideoId)=(["\'])(?P<code>(?:(?!\1).)+)\1', webpage)]
+        asset_id = traverse_obj(direct_info, ('video', 'publicId', {str}))
+        if not asset_id:
+            self.raise_no_formats('Could not find any video formats', video_id=video_id)
 
-        title = self._html_search_regex(
-            r'(?s)<(?:span|div)[^>]+class="(?:title|wrapper)"[^>]*>.*?<h1>(.+?)</h1>',
-            webpage, 'title', fatal=False)
+        result = {
+            'display_id': video_id,
+            'formats': self._extract_m3u8_formats(
+                self._create_asset_url(f'/video/upload/sp_full_hd/v1/{asset_id}.m3u8'), video_id),
+        }
+        result.update(traverse_obj(direct_info, {
+            'id': ('id', {str}),
+            'title': ('name', {str}),
+            'timestamp': ('startDate', {unified_timestamp}),
+            'description': ('description', 'text', {str}),
+            'age_limit': ('contentRating', 'order', {int}),
```

Comment:
This seems too direct, and gives `17` for the first test. Should we use some of the other indicators perhaps?